### PR TITLE
Restyle error messages

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- As I was working on #2862 I was bothered by the Vite errors that we were exposing to Astro users.
- This PR restyles our error messages with a fresh can of paint
- more importantly, it sets up some handling for re-phrasing Vite error messages that we can add to going forward.

#### Build (with metadata)
Screen Shot 2022-03-26 at 3 12 43 PM

#### Dev  (with metadata)
Screen Shot 2022-03-26 at 3 12 52 PM

#### Without metadata

Screen Shot 2022-03-26 at 3 15 33 PM

## Testing

- Error output tested manually

## Docs

- N/A

### CODE REVIEW

File: packages/astro/src/core/build/index.ts [hunk 1]
```diff
@@ -35,32 +38,30 @@ class AstroBuilder {
 	private origin: string;
 	private routeCache: RouteCache;
 	private manifest: ManifestData;
-	private viteServer?: vite.ViteDevServer;
-	private viteConfig?: ViteConfigWithSSR;
+	private timer: Record<string, number>;
 
 	constructor(config: AstroConfig, options: BuildOptions) {
-		applyPolyfill();
-
```
Comment (natemoo-re) [inline packages/astro/src/core/build/index.ts:43]: Was this intentional?

**Review by natemoo-re — APPROVED** [review]
Clean refactors, CLI output looks great. ✅

File: packages/astro/src/core/build/index.ts (hunk 1 above)
Comment (FredKSchott) [reply packages/astro/src/core/build/index.ts:43]: Yea, kind of unrelated but this moves `applyPolyfill()` into the CLI function that gets called (`add()`, `build()`, etc) to match all of the other commands. 

Also it's just a bit odd to have a side-effect like this happen in the constructor of another object., imo

File: packages/astro/src/core/build/generate.ts [hunk 2]
```diff
@@ -67,7 +67,8 @@ export function chunkIsPage(astroConfig: AstroConfig, output: OutputAsset | Outp
 }
 
 export async function generatePages(result: RollupOutput, opts: StaticBuildOptions, internals: BuildInternals, facadeIdToPageDataMap: Map<string, PageBuildData>) {
-	info(opts.logging, null, `\n${bgMagenta(black(' generating static routes '))}\n`);
+	const timer = performance.now();
+	info(opts.logging, null, `\n${bgGreen(black(' generating html '))}`);
```
Comment (bholmesdev) [inline packages/astro/src/core/build/generate.ts:71]: This isn't _always_ going to be HTML if you have JSON routes! May prefer the original wording for clarity

```suggestion
	info(opts.logging, null, `\n${bgGreen(black(' generating static routes '))}`);
```

File: packages/astro/src/core/build/generate.ts [hunk 3]
```diff
@@ -175,65 +164,61 @@ async function generatePath(pathname: string, opts: StaticBuildOptions, gopts: G
...
-			if (result.type !== 'html') {
-				return;
-			}
-			body = result.html;
+	let body: string;
+	if (pageData.route.type === 'endpoint') {
+		const result = await callEndpoint(mod as unknown as EndpointHandler, options);
+
+		if (result.type === 'response') {
+			throw new Error(`Returning a Response from an endpoint is not supported in SSG mode.`);
```
Comment (bholmesdev) [inline packages/astro/src/core/build/generate.ts:207]: Oh good catch!

File: packages/astro/src/core/build/index.ts (hunk 1 above)
Comment (natemoo-re) [reply packages/astro/src/core/build/index.ts:43]: That's fair!

File: packages/astro/src/core/build/generate.ts (hunk 2 above)
Comment (FredKSchott) [reply packages/astro/src/core/build/generate.ts:71]: good call!

File: packages/astro/src/core/build/generate.ts (hunk 3 above)
Comment (FredKSchott) [reply packages/astro/src/core/build/generate.ts:207]: I wish I came up with this! but it was just moved from elsewhere. You probably have @matthewp to thank :)

**Pull request merged**